routes/eth.go

```diff
@@ -241,41 +260,124 @@ func (fes *APIServer) SubmitETHTx(ww http.ResponseWriter, req *http.Request) {
 	}
 
+	res := SubmitETHTxResponse{
+		BitCloutTxnHash: bitcloutTxnHash.String(),
+	}
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Problem encoding response: %v", err))
+		return
+	}
+}
+
+// 1. Validate the transaction mined
+// 2. Calculate the nanos to send
+// 3. Send the nanos
+// 4. Record the successful send
+func (fes *APIServer) finishETHTx(ethTxLog *ETHTxLog) (txnHash *lib.BlockHash, _err error) {
+	// Ensure the transaction mined
+	if ethTxLog.Tx.BlockHeight < 0 {
+		return nil, errors.New("Transaction failed to mine")
+	}
+
 	// Fetch buy bitclout basis points fee
 	feeBasisPoints, err := fes.GetBuyBitCloutFeeBasisPointsResponseFromGlobalState()
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Error getting buy fee basis points: %v", err))
-		return
+		return nil, errors.New(fmt.Sprintf("Error getting buy fee basis points: %v", err))
 	}
 
 	// Calculate nanos purchased
-	totalWei := big.NewFloat(0).SetInt(ethTx.Total)
+	totalWei := big.NewFloat(0).SetInt(ethTxLog.Tx.Total)
 	totalEth := big.NewFloat(0).Quo(totalWei, big.NewFloat(1e18))
 	nanosPurchased := fes.GetNanosFromETH(totalEth, feeBasisPoints)
 
-	bitcloutTxnHash, err := fes.SendSeedBitClout(pkBytes, nanosPurchased, true)
+	bitcloutTxnHash, err := fes.SendSeedBitClout(ethTxLog.PublicKey, nanosPurchased, true)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error sending BitClout: %v", err))
+	}
+
+	// Record successful transaction in global state
+	ethTxLog.BitCloutTxnHash = bitcloutTxnHash.String()
+	globalStateKey := GlobalStateKeyETHPurchases(ethTxLog.Tx.Hash)
+	globalStateVal := bytes.NewBuffer([]byte{})
+	err = gob.NewEncoder(globalStateVal).Encode(ethTxLog)
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Error sending BitClout: %v", err))
+		return nil, errors.New(fmt.Sprintf("Failed to encode ETH transaction: %v", err))
+	}
+
+	err = fes.GlobalStatePut(globalStateKey, globalStateVal.Bytes())
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error processing GlobalStatePut: %v", err))
+	}
+
+	return bitcloutTxnHash, nil
+}
+
+type AdminProcessETHTxRequest struct {
+	ETHTxHash string
+}
+
+type AdminProcessETHTxResponse struct {
+	BitCloutTxnHash string
+}
+
+func (fes *APIServer) AdminProcessETHTx(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := AdminProcessETHTxRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminProcessETHTx: Problem parsing request body: %v", err))
 		return
 	}
 
-	// Record transaction success in global state
-	if err = fes.GlobalStatePut(globalStateKey, []byte{1}); err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Error processing GlobalStatePut: %v", err))
+	if !fes.IsConfiguredForETH() {
+		_AddBadRequestError(ww, "AdminProcessETHTx: Not configured for ETH")
```

Comment:
I would move this ahead of the decoding since we won't even want to bother decoding the payload if the node is not configured for ETH